pyiron_contrib/RDM/storagejob.py

```diff
@@ -0,0 +1,324 @@
+import os
+import posixpath
+import shutil
+
+from pyiron_base import GenericJob, DataContainer
+from pyiron_base.generic.filedata import FileData
+from pyiron_contrib.generic.s3io import FileS3IO
+
+
+def _remove_s3_working_directory(file_s3_io_handle, group_to_remove):
+    # Upon moving s3 to base, this should be moved to pyiron_base/job/util.py
+    if file_s3_io_handle.is_dir(group_to_remove):
+        file_s3_io_handle.remove_group(path=group_to_remove)
+
+
+class StorageType(DataContainer):
+    def __init__(self, *, init=None, storage_type=None, table_name=None):
+        """What type of storage is used
+
+        Parameters:
+            storage_type(str): one of ['local', 's3']
+        """
+        super().__init__(init=init, table_name=table_name)
+        self._available_storage_types = ['local', 's3']
+        self._storage_type = None
+        self.storage_type = storage_type
+
+    def create_group(self, name):
```

Comment:
IMO this method should be pushed up to `DataContainer`. Your thoughts @pmrv?

Reply:
`create_group` actually lives already on `DataContainer`. However, I need to overwrite this since by default the new group is of the same class (`self.__class__()`) as the original container. Here, I want to have a plain `DataContainer` without my restrictions as a sub group.

Reply:
> `create_group` actually lives already on `DataContainer`. 

Ah, so it does! My apologies, I thought I checked for that and found it missing but I must have been looking at the wrong class...